dataset-1-4-1/feature_defs.json

```diff
@@ -107,22 +125,142 @@
+        "tooltip": "Name of the cellular structure that has been fluorescently labeled in each cell line",
+        "discrete": true,
+        "options": {
+            "AICS-5": {
+                "color": "#77207C",
+                "name": "Matrix adhesions",
+                "key": "Paxillin"
+            },
+            "AICS-7": {
+                "color": "#FF96FF",
+                "name": "Actin filaments",
+                "key": "Alpha-actinin-1"
+            },
+            "AICS-10": {
+                "color": "#FFFFB5",
+                "name": "Endoplasmic reticulum",
+                "key": "Sec61 beta"
+            },
+            "AICS-11": {
+                "color": "#FFD184",
+                "name": "Mitochondria",
+                "key": "Tom20"
+            },
+            "AICS-12": {
+                "color": "#6B4500",
+                "name": "Microtubules",
+                "key": "Alpha-tubulin"
+            },
+            "AICS-13": {
+                "color": "#084AE7",
+                "name": "Nuclear envelope",
+                "key": "Lamin B1"
+            },
+            "AICS-14": {
+                "color": "#A9D1E5",
+                "name": "Nucleolus (Dense Fibrillar Component)",
+                "key": "Fibrillarin"
+            },
+            "AICS-16": {
+                "color": "#FFB1FF",
+                "name": "Actin filaments",
+                "key": "Beta-actin"
+            },
+            "AICS-17": {
+                "color": "#A850E8",
+                "name": "Desmosomes",
+                "key": "Desmoplakin"
+            },
+            "AICS-22": {
+                "color": "#AD952A",
+                "name": "Lysosome",
+                "key": "LAMP-1"
+            },
+            "AICS-23": {
+                "color": "#A850D4",
+                "name": "Tight junctions",
+                "key": "Tight junction ZO-1"
+            },
+            "AICS-24": {
+                "color": "#FF82FF",
+                "name": "Actomyosin bundles",
+                "key": "Non-muscle myosin heavy chain IIB"
+            },
+            "AICS-25": {
+                "color": "#B7952A",
+                "name": "Golgi",
+                "key": "Sialyltransferase 1"
+            },
+            "AICS-32": {
+                "color": "#9D7000",
+                "name": "Centrosome",
+                "key": "Centrin-2"
+            },
+            "AICS-33": {
+                "color": "#FFD16E",
+                "name": "Peroxisomes",
+                "key": "Peroxisomal membrane protein PMP34"
+            },
+            "AICS-40": {
+                "color": "#FFD150",
+                "name": "Endosomes",
+                "key": "Ras-related protein Rab-5A"
+            },
+            "AICS-53": {
+                "color": "#A850C0",
+                "name": "Gap junctions",
+                "key": "Connexin-43"
```

Comment:
Add "key" to the options schema

Reply:
perhaps, "customData", "details", "userData", "info", "altName"?

Reply:
@toloudis, @schoinh and I were discussing what would be a better term for this. It's just supposed to be a catchall for additional data we might want on the front end. For cell-line this is pretty critical information, we're using protein name in the thumbnail cards and in the gallery, and we're showing it in the tooltip when you roll over the structure in the checkbox menu.

Reply:
`key` doesn't feel right because it's not really a key, right?  to me, `userData` and `customData` imply objects with their own specific properties.  I kind of like `alt` because it shows up in web HTML.   `altName` or `alternate` or `alternateName` also sound reasonable.

Reply:
what do we do if there's a third name like gene name? maybe make `name` an array?  or just say right now that there can only be two, main and alternate

Reply:
When we discussed it more I realized we might need both. In this case protein is a key in that it's the most specific information (since more than one protein can label the same structure). But we might also want additional information that isn't a key, so maybe a schema that looks like: 
```
color: string;
name: string;
key?: string;
altName?: string;
```

Reply:
I'm still a little confused by the use of `key` here.  Isn't the key really the AICS-## cell line id from our DB?  Also, if we don't use that AICS-## string anywhere, then do the numbers here matter? I mean why wouldn't they just be an incremented integer index?   If `name` is structure name, `key` is protein name, and `altName` might be gene name, then why not just use name1, name2, name3 or something.  aren't they kind of interchangeable?  I'm not seeing how `key` would be used to do anything different than `name` would.

Reply:
because name isn't required to be unique. More than one protein can label the same structure, so name/displayname can be a more general thing, while 'key' is something that is a unique identifier. The reason for this overall is that we're displaying the "groupby" data in the card over the plot, so the value needs to be in the condensed json on AWS, so I used 'key' if it exists and 'name' if key doesn't exist.

Reply:
https://github.com/allen-cell-animated/cell-feature-data/pull/27/files#diff-fc0fe9fd9f524c21a77127adcf029d3b37dcbc1a7078865dd4d24f0b6a453c9bR46